server/liveblog/blogs/tasks.py

```diff
@@ -36,37 +37,92 @@ def publish_embed(blog_id, api_host=None, theme=None):
     return superdesk.upload.url_for_media(file_id)
 
 
-def delete_embed(blog_id):
+def delete_embed(blog_id, theme=None, output=None):
     check_media_storage()
-    file_path = get_blog_path(blog_id)
+    blog = get_resource_service('client_blogs').find_one(req=None, _id=blog_id)
+    public_urls = blog.get('public_urls', {'output': {}, 'theme': {}})
+    if output:
+        output_id = str(output.get('_id'))
+        public_url = public_urls['output'][output_id]
+        public_urls['output'].pop(output_id)
+    elif theme:
+        public_url = public_urls['theme'][theme]
+        public_urls['theme'].pop(theme)
+    else:
+        for output_url in public_urls['outputs']:
+            app.media.delete(output_url)
+        for theme_url in public_urls['theme']:
+            app.media.delete(theme_url)
+        public_url = blog.get('public_url')
+        public_urls = {}
+
     # Remove existing file.
-    app.media.delete(file_path)
+    app.media.delete(public_url)
+    get_resource_service('blogs').system_update(blog_id, {'public_urls': public_urls}, blog)
 
 
-def _publish_blog_embed_on_s3(blog_id, safe=True):
+def _publish_blog_embed_on_s3(blog_id, theme=None, output=None, safe=True):
     blog = get_resource_service('client_blogs').find_one(req=None, _id=blog_id)
+
+    # get the `output` data if the `output_id` is set.
+    # if output and isinstance(output, str):
+    #     output = get_resource_service('outputs').find_one(req=None, _id=output)
+    output_id = None
+    if output:
+        # get the output `_id`
+        output_id = str(output.get('_id'))
+
+        # compile a theme if there is an `output`.
+        if output.get('theme'):
+            theme = output.get('theme')
+        else:
+            theme = blog['blog_preferences'].get('theme')
+
     if blog['blog_preferences'].get('theme', False):
         try:
-            public_url = publish_embed(blog_id, '//%s/' % (app.config['SERVER_NAME']))
-            get_resource_service('blogs').system_update(blog_id, {'public_url': public_url}, blog)
-            push_notification('blog', published=1, blog_id=blog_id, public_url=public_url)
+            public_url = publish_embed(blog_id,
+                                       theme,
+                                       output,
+                                       api_host='//{}/'.format(app.config['SERVER_NAME']))
+            public_urls = blog.get('public_urls', {'output': {}, 'theme': {}})
+
+            if output_id:
+                public_urls['output'][output_id] = public_url
+                get_resource_service('blogs').system_update(blog_id, {'public_urls': public_urls}, blog)
+                push_notification('blog', published=1, blog_id=blog_id, public_urls=public_urls)
+            elif theme:
```

Comment:
Again, not clear in the method signature that you define either the theme or the output but not both.